pkg/k8shandler/kibana/reconciler.go

```diff
@@ -393,6 +391,27 @@ func updateCurrentDeploymentImages(current *apps.Deployment, desired *apps.Deplo
 	return current
 }
 
+func updateCurrentDeploymentEnvIfDifferent(current *apps.Deployment, desired *apps.Deployment) bool {
+
+	different := false
+
+	containers := current.Spec.Template.Spec.Containers
+
+	for index, curr := range current.Spec.Template.Spec.Containers {
+		for _, des := range desired.Spec.Template.Spec.Containers {
+			// Only compare the env of containers with the same name
+			if curr.Name == des.Name {
+				if !utils.EnvValueEqual(curr.Env, des.Env) {
+					containers[index].Env = des.Env
```

Comment:
Is this a Go-ism I don't get?

Reply:
sort of, it won't let you update the object that you are currently iterating over. so we have to access it via an index from the original array

Reply:
Looked it up, thanks! Not that this is hot code or anything, but if we just go with indices we avoid copying the large v1.Container structs in addition to sticking to only one way of accessing the slice elements:
```
	different := false
	
	for i := range current.Spec.Template.Spec.Containers {
		curr := &current.Spec.Template.Spec.Containers[i]
		for j := range desired.Spec.Template.Spec.Containers {
			des := &desired.Spec.Template.Spec.Containers[j]
			// Only compare the env of containers with the same name
			if curr.Name == des.Name {
				if !utils.EnvValueEqual(curr.Env, des.Env) {
					curr.Env = des.Env
					different = true
				}
			}
		}
	}

	return different
```

Reply:
i'm not convinced we wouldn't still be scope confined here...
do you mind if we address this in a tech debt follow up? we have this pattern in a few other places that we should clean it all up if we are going to address this...

Reply:
It's your PR, I'm only offering suggestions :)
Unit tests in this kibana package pass with this version of the code. I found this to be a good explanation https://stackoverflow.com/questions/15945030/change-values-while-iterating